zebra/zebra_mlag.c

```diff
@@ -88,8 +878,557 @@ void zebra_mlag_init(void)
 {
 	install_element(VIEW_NODE, &show_mlag_cmd);
 	install_element(ENABLE_NODE, &test_mlag_cmd);
+	install_element(ENABLE_NODE, &test_mlag_route_cmd);
+	install_element(ENABLE_NODE, &test_mlag_route_bulk_cmd);
+
+	/*
+	 * Intialiaze teh MLAG Global variableis
+	 * write thread will be craeted during actual registration with MCLAG
+	 */
+	zrouter.mlag_info.clients_interested_cnt = 0;
+	zrouter.mlag_info.connected = false;
+	zrouter.mlag_info.timer_running = false;
+	zrouter.mlag_info.mlag_fifo = stream_fifo_new();
+	zrouter.mlag_info.zebra_pth_mlag = NULL;
+	zrouter.mlag_info.th_master = NULL;
+	zrouter.mlag_info.t_read = NULL;
+	zrouter.mlag_info.t_write = NULL;
+	test_mlag_in_progress = false;
+	zebra_mlag_reset_write_buffer();
+	zebra_mlag_reset_read_buffer();
 }
 
 void zebra_mlag_terminate(void)
 {
 }
+
+
+/*
+ *
+ *  ProtoBuf Encoding APIs
+ */
+
+#ifdef HAVE_PROTOBUF
+
+DEFINE_MTYPE_STATIC(ZEBRA, MLAG_PBUF, "ZEBRA MLAG PROTOBUF")
+
+int zebra_mlag_protobuf_encode_client_data(struct stream *s, uint32_t *msg_type)
+{
+	ZebraMlagHeader hdr = ZEBRA_MLAG__HEADER__INIT;
+	struct mlag_msg mlag_msg;
+	uint8_t tmp_buf[ZEBRA_MLAG_BUF_LIMIT];
+	int len = 0;
+	int n_len = 0;
+	int rc = 0;
+	char buf[80];
```

Comment:
please don't use the literal size here? use a macro (and re-use it below, where the size of `buf` is needed, or use `sizeof(buf)`)

Reply:
Done